src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Sockets.Tests
+{
+    public class UnixDomainSocketTest : IDisposable
+    {
+        private readonly ITestOutputHelper _log;
+        private readonly string _path;
+        private readonly UnixDomainSocketEndPoint _endPoint;
+
+        public UnixDomainSocketTest(ITestOutputHelper output)
+        {
+            _log = TestLogging.GetInstance();
+            _path = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            if (File.Exists(_path))
+            {
+                File.Delete(_path);
+            }
+
+            _endPoint = new UnixDomainSocketEndPoint(_path);
+        }
+
+        private void OnConnectAsyncCompleted(object sender, SocketAsyncEventArgs args)
+        {
+            ManualResetEvent complete = (ManualResetEvent)args.UserToken;
+            complete.Set();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Socket_CreateUnixDomainSocket_Throws_OnWindows()
+        {
+            Assert.False(Capability.UnixDomainSocketsSupport());
+            Assert.Throws<SocketException>(() => new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified));
```

Comment:
The behavior this test is covering is not new and has not been changed by the pull request. Technically, this test is not needed, but I wanted to document the behavior of Socket(AddressFamily.Unix,...) on Windows. Given that there is no change in the behavior, and it's common to check the exception type and not the message in the code, the test only checks for the exception type.

I believe the exception comes from the socket pal layer which in turn calls winsock on Windows. What 3-rd party lib are you referring to?